### ISSUE TITLE
Config pages not refreshing after settings update

### ISSUE THREAD

**Issue by ulchm** [SickRage/sickrage-issues#471]
Currently if I edit the settings on my config/postprocessing page, pressing save doesn't refresh the page. 

I spoke with Labrys on IRC about this moments ago and he explained that the expected behaviour is a page refresh after load (I was inquiring about some sort of a confirmation message).

Branch: Master
OS:  Ubuntu 14.04

**Comment by SickRage2** [comment]
Thanks for the issue report! Before a real human comes by, please make sure your report has all the below criteria checked

- [ ] Include basic information: Branch/Commit, OS, What you did, What happened, What you expected
- [ ] Enable debug logging (be sure to disable after the bug is fixed)
- [ ] Post debug logs, either inline (for smaller logs) or using gist)

Please make sure you also read [how to create an issue](https://github.com/SickRage/sickrage-issues/blob/master/README.md) and followed all of the steps. 

The title should describe your issue. Having "SR not working" or "I get this bug" for 100 issues, isn't really helpful. We will close issues if there isn't enough information.

Sometimes the devs may seem like grunts and respond with short answers. This isn't (always) because the dev hates you, but because he's on mobile or busy fixing bugs. If something isn't clear, please let us know, and this bot may get updated to automatically answer you.

Thanks!

**Comment by OmgImAlexis** [comment]
@labrys why should the page refresh? If it saved shouldn't there just be a web notification that the config.ini was saved?

**Comment by OmgImAlexis** [comment]
Clicking the save config button should just show a notification like so. I don't see a reason why the page should refresh?

![screen shot 2016-01-12 at 11 31 39 pm](https://cloud.githubusercontent.com/assets/6525926/12264166/ac808dac-b984-11e5-82f5-2905a4b1e29b.png)

**Comment by labrys** [comment]
@OmgImAlexis Many people disable popup notifications which means there is no obvious indication that the action has been submitted.  In addition some pages such as the manage searches pages should refresh to show the change in status.  While a refresh isn't particularly necessary, an indication of some sort (other than a popup) is required.

**Comment by OmgImAlexis** [comment]
@labrys if they turn off notifications then that's on them. I don't see a reason to have this page inparticular refresh when there's no need to. If they want an indication as to it saving then they need notifications turned on.

Closing this for now as I don't see it as a bug as the expected behavior is happening. If there are other pages like you said you should open another issue on them. This is correct as it is.

**Comment by fernandog** [comment]
@OmgImAlexis  

what about this?
" addition some pages such as the manage searches pages should refresh to show the change in status.

**Comment by miigotu** [comment]
The solution wouldnt be a page refresh either way, it would be the js hding/showing elements that should now be enabled/disabled and updating values.

For the post processing page this isnt even necessary.

I agree that if they are blocking web notifications they cant expect a web notification.

**Comment by fernandog** [comment]
ok but before we do that, we should add back the refresh, don't you agree?

**Comment by miigotu** [comment]
sure, maybe we should go back to 1999 also?

**Comment by fernandog** [comment]
nice year lol

**Comment by OmgImAlexis** [comment]
@fernandog the only reason they refreshed was the js was setup wrong and didn't block the page refreshing. We shouldn't refreshing anything unless absolutely necessary and even then we can usually fake it by just fetching the div needed with ajax and replacing it on the page.

**Comment by labrys** [comment]
@miigotu @OmgImAlexis Web notifications is not a proper way to display form input.  A proper way would be like we do for testing connections to the NZB/Torrent clients.  Its a matter of effective user interface design.

**Comment by miigotu** [comment]
@labrys the web notification is not to display the form input, but that it actually submitted. The values just need refreshed, not the entire page

**Comment by labrys** [comment]
I don't disagree about the page not needing refreshed, but if notifications are blocked the user gets no indication of a submitted response.  Notifications are not sufficient to show something has been accomplished.

**Comment by OmgImAlexis** [comment]
@labrys blocking notifications isn't a recommended thing though and honestly I don't see why they'd turn them off. If the user wants to see things like this they just need to turn them on and use SickRage in the expected manner instead of disabling them. If we change this people will expect other things we've said no to for this exact reason.

**Comment by fernandog** [comment]
can we show a div showing users need to enable notifications for SR?
some users may have notifications disabled for all sites. not only SR. and not all of them know about that.

**Comment by OmgImAlexis** [comment]
@fernandog if they've got them disabled for all sites they should expect to not get any notifications, if they have them just disabled for SickRage then they they should know what they're doing. This is end of discussion, we can't keep changing thing for people that want to use SickRage in an unsupported way.

**Comment by labrys** [comment]
So if you placed an order on e.g. amazon and submitted it you think it would be acceptable to for them to only provide confirmation in a notification that may be blocked by a browser?  I think not.

**Comment by OmgImAlexis** [comment]
@labrys that's two completely different things. If you save your settings on most sites now you don't even get any notification, hell on Facebook most of the time you save and just get redirected to the homepage.

I work on the js, if it gets changed to "fix" something that isn't an issue I'll just roll it back.

**Comment by duramato** [comment]
@OmgImAlexis I agree with @labrys , you cant rely on notifications that dont even work on some browsers/systems , besids the user can even disable them..

**Comment by labrys** [comment]
A redirect to the home page is a confirmation.. it wouldnt have redirected you if it didnt get submitted.

**Comment by labrys** [comment]
http://www.usability.gov/what-and-why/user-interface-design.html
> Make sure that the system communicates what’s happening.  Always inform your users of location, actions, changes in state, or errors. The use of various UI elements to communicate status and, if necessary, next steps can reduce frustration for your user. 

https://developer.apple.com/library/ios/documentation/UserExperience/Conceptual/MobileHIG/InteractivityInput.html#//apple_ref/doc/uid/TP40006556-CH55-SW1
>Feedback Aids Understanding
>
>Feedback helps users know what an app is doing, discover what they can do next, and understand the results of their actions. UIKit controls and views provide many kinds of feedback.

>As much as possible, integrate status and other relevant feedback information into your UI. It’s best when users can get this type of information without taking action or being distracted from their content. For example, Mail displays the current mailbox status in the toolbar where it doesn’t compete with the user’s content.

> Avoid unnecessary alerts. An alert is a powerful feedback mechanism, but it should be used only to deliver important—and ideally actionable—information. If users see too many alerts that don’t contain essential information, they quickly learn to ignore all alerts. To learn more about using an alert, see Alert.
